### PR TITLE
x

### PR DESCRIPTION
**Pull request by metacom27**
x

### CODE REVIEW

**Comment by bcoles** [comment]
It is required that code in your fork be merged **from a unique branch** in your repository to master in Rapid7's.  Please create a new branch in your fork of framework and resubmit this from that branch.

```
git checkout -b <BRANCH_NAME>
git push <your_fork_remote> <BRANCH_NAME>
```

This helps protect the process, ensure users are aware of commits on the branch being considered for merge, allows for a location for more commits to be offered without mingling with other contributor changes and allows contributors to make progress while a PR is still being reviewed.

Closing based on the this requirement, please do resubmit from a unique branch.

File: modules/exploits/windows/fileformat/xinfire_dvd_player.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
```
Comment (bcoles) [inline modules/exploits/windows/fileformat/xinfire_dvd_player.rb:2]:
```suggestion
# This module requires Metasploit: https://metasploit.com/download
```

File: modules/exploits/windows/fileformat/xinfire_dvd_player.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class MetasploitModule < Msf::Exploit::Remote
+  #Rank definition: http://dev.metasploit.com/redmine/projects/framework/wiki/Exploit_Ranking
+  #ManualRanking/LowRanking/AverageRanking/NormalRanking/GoodRanking/GreatRanking/ExcellentRanking
```
Comment (bcoles) [inline modules/exploits/windows/fileformat/xinfire_dvd_player.rb:10]:
```suggestion
```

File: modules/exploits/windows/fileformat/xinfire_dvd_player.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
```
Comment (bcoles) [inline modules/exploits/windows/fileformat/xinfire_dvd_player.rb:6]:
```suggestion
```

File: modules/exploits/windows/fileformat/xinfire_dvd_player.rb [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+  include Msf::Exploit::FILEFORMAT
+  include Msf::Exploit::Seh
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'    => 'Xinfire DVD Player',
+      'Description'  => %q{
+        The vulnerability is triggered when opening a malicious PLF file.How it works:open plf file on (playlist), and the vulnerability is triggered.
+		This software is similar as DVD X Player and BlazeDVD.This module has been tested successfully on Win7, Win10.
```
Comment (bcoles) [inline modules/exploits/windows/fileformat/xinfire_dvd_player.rb:21]: Please wrap to 80 or 120 columns

File: modules/exploits/windows/fileformat/xinfire_dvd_player.rb [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+          [ ' Xinfire DVD Player and Pro v.5.5.0.0 ',
+            {
+              'Ret'     =>  0x6031ec94, # pop eax # pop esi # ret  - Configuration.dll
+              'Offset'  =>  608
+            }
+          ],
+        ],
+      'Privileged'  => false,
+      #Correct Date Format: "M D Y"
+      #Month format: Jan,Feb,Mar,Apr,May,Jun,Jul,Aug,Sep,Oct,Nov,Dec
```
Comment (bcoles) [inline modules/exploits/windows/fileformat/xinfire_dvd_player.rb:60]:
```suggestion
```

File: modules/exploits/windows/fileformat/xinfire_dvd_player.rb [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+      'DisclosureDate'  => 'Apr 16 2020',
+      'DefaultTarget'  => 0))
+
+    register_options([OptString.new('FILENAME', [ false, 'The file name.', 'msf.plf']),], self.class)
```
Comment (bcoles) [inline modules/exploits/windows/fileformat/xinfire_dvd_player.rb:64]:
```suggestion
    register_options([OptString.new('FILENAME', [ false, 'The file name.', 'msf.plf'])])
```

File: modules/exploits/windows/fileformat/xinfire_tv_player.rb [hunk 7]
```diff
@@ -0,0 +1,77 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
```
Comment (bcoles) [inline modules/exploits/windows/fileformat/xinfire_tv_player.rb:2]:
```suggestion
# This module requires Metasploit: https://metasploit.com/download
```

File: modules/exploits/windows/fileformat/xinfire_tv_player.rb [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class MetasploitModule < Msf::Exploit::Remote
+  #Rank definition: http://dev.metasploit.com/redmine/projects/framework/wiki/Exploit_Ranking
+  #ManualRanking/LowRanking/AverageRanking/NormalRanking/GoodRanking/GreatRanking/ExcellentRanking
```
Comment (bcoles) [inline modules/exploits/windows/fileformat/xinfire_tv_player.rb:10]:
```suggestion
```

File: modules/exploits/windows/fileformat/xinfire_tv_player.rb [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+
+  include Msf::Exploit::FILEFORMAT
+  include Msf::Exploit::Seh
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'    => 'Xinfire TV Player',
+      'Description'  => %q{
+        The vulnerability is triggered when opening a malicious PLF file.How it works:open plf file on (playlist), and the vulnerability is triggered.
+		This software is similar as Aviosoft Digital TV Player and BlazeVideo HDTV Player.This module has been tested successfully on Win7, Win10.
```
Comment (bcoles) [inline modules/exploits/windows/fileformat/xinfire_tv_player.rb:21]: Please line wrap to 80 or 120 columns

File: modules/exploits/windows/fileformat/xinfire_tv_player.rb [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+            }
+          ],
+        ],
+      'Privileged'  => false,
+      #Correct Date Format: "M D Y"
+      #Month format: Jan,Feb,Mar,Apr,May,Jun,Jul,Aug,Sep,Oct,Nov,Dec
+      'DisclosureDate'  => 'Apr 15 2020',
+      'DefaultTarget'  => 0))
+
+    register_options([OptString.new('FILENAME', [ false, 'The file name.', 'msf.plf']),], self.class)
```
Comment (bcoles) [inline modules/exploits/windows/fileformat/xinfire_tv_player.rb:63]:
```suggestion
    register_options([OptString.new('FILENAME', [ false, 'The file name.', 'msf.plf'])])
```

File: modules/exploits/windows/fileformat/xinfire_tv_player.rb [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+        [
+          'All the guys which found the vulnerability in Aviosoft Digital TV Player and BlazeVideo HDTV Player',  # Original discovery
+		  'metacom27@gmail.com - twitter.@m3tac0m',
+          'metacom',  # MSF Module
+        ],
+      'References'  =>
+        [
+          [ 'OSVDB', '77043' ],
+          [ 'CVE', '2009-0450, 2006-6396, 2006-6199' ],
+          [ 'URL', '<insert another link to the exploit/advisory here>' ]
```
Comment (adfoster-r7) [inline modules/exploits/windows/fileformat/xinfire_tv_player.rb:34]: 👀

File: modules/exploits/windows/fileformat/xinfire_dvd_player.rb [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+        [
+          'All the guys which found the vulnerability in DVD X Player and BlazeDVD',  # Credits - Original discovery
+		  'E-mail:metacom27@gmail.com - twitter.@m3tac0m',
+          'metacom',  # MSF Module
+        ],
+      'References'  =>
+        [
+          [ 'OSVDB','36956' ],
+          [ 'CVE', '2007-3068' ],
+          [ 'URL', '<insert another link to the exploit/advisory here>' ]
```
Comment (adfoster-r7) [inline modules/exploits/windows/fileformat/xinfire_dvd_player.rb:35]: 👀

**Comment by metacom27** [comment]
ok

**Comment by bcoles** [comment]
#13288

**Pull request closed without merging**